ReactWindows/ReactNative.Shared/Modules/Network/DefaultHttpClient.cs

```diff
@@ -35,6 +40,11 @@ public DefaultHttpClient(HttpClient client)
                     throw;
                 }
             }
+#else
+            var asyncInfo = _client.SendAsync(request);
+            asyncInfo.Wait(token);
```

Comment:
Okay I double checked.  System.Net.Http.HttpClient.SendAsync returns Task<T>.  Task<T>.Wait is 100% a blocking operation.  This is not what you want to do here.

Instead you want to:
```c#
return await _client.SendAsync(request, token).ConfigureAwait(false); 
```

It's unfortunate that because of the code sharing, you can't remove the `async` keyword from the method, otherwise you could just:

```c#
return _client.SendAsync(request, token);
```

---
In reply to: [83638663](https://github.com/ReactWindows/react-native-windows/pull/787#discussion_r83638663) [](ancestors = 83638663)